esp-hal/src/timer/systimer.rs

```diff
@@ -463,7 +430,13 @@ where
             .bit_is_set();
 
         let us = value.ticks();
-        let ticks = us * (SystemTimer::TICKS_PER_SECOND / 1_000_000) as u64;
+        let ticks = us * (SystemTimer::TICKS_PER_SECOND / 1_000_000);
+
+        // The `SYSTIMER_TARGETx_PERIOD` field is 26-bits wide, so we must
```

Comment:
iirc from the trm, it's 26bit for when in periodic mode, but up to 52bit for alarm/oneshot mode, so I think we need two separate checks based on the mode.

Reply:
I think also the one-shot target bit-count is different for S2 - and also for periodic it seems like 26 vs 30 bits